### DISCUSSION TITLE
Performance Challenges of Blazor Server Applications in Low Bandwidth Environments Compared to MVC and Traditional Web Applications

### DISCUSSION THREAD

**Discussion by neozhu** [dotnet/aspnetcore#55693] [General]
## Background and Motivation

This topic explores the performance issues encountered with Blazor Server applications when operated in environments with limited bandwidth. It specifically focuses on significant response delays that become apparent during user interactions, such as clicking buttons to trigger events. The discussion aims to compare these experiences with those of MVC and traditional web applications, highlighting the severity and impact of latency in Blazor Server applications under such conditions.

**Comment by neozhu** [comment]
please try on it
https://architecture.blazors.app/

**Comment by mkArtakMSFT** [comment]
@neozhu can you share more about your setup and what are your metrics? it would be great to know what is being measured and how it differs between the two stacks that you compare.

**Comment by garrettlondon1** [comment]
Blazor Server applications do not use Javascript for interactions, so every roundtrip is sent over websocket. If there is latency, the websocket will be slow.

The reason that this is not noticable in MVC is because the buttons/etc, are not hitting the server for interactivity on click. It is a known tradeoff of Blazor Server.

They are hitting the server once the MVC request goes out, but the feedback from buttons, does not
